lib/part.mli

```diff
@@ -38,15 +38,14 @@ val replace : file -> part:string option -> lines:string list -> file
 (* Exposed for test purposes only *)
 
 module Parse_parts : sig
-  type part_decl =
-    | Normal of string
+  type t =
+    | Content of string
     | Compat_attr of string * string
     (* ^^^^ This is for compat with the [[@@@part name]] delimiters *)
     | Part_begin of string * string
-    | Part_end
-    | File_end
+    | Part_end of string option
```

Comment:
We could probably use an inline record here to make it explicit, e.g.:
```suggestion
    | Part_end of {inline_delimiter_prefix: string option}
```
or any field name that properly describe what this string is.

What do you think?

Reply:
Despite implementing it like this initially, I'm currently trying to get rid of this again, because I don't find the model to be good.

There is `Normal`/`Content` `of string` which contains the text of the segment and there's the markers which contain metadata about the delimiters. Forcing a bit of `content` onto the `Part_end` feels wrong to me, this was done because `parse_line` is just returning one item per line - but as we have seen a line can (but does not have to) have both `Content` and `Part_end`, so the function should just return a (most of the time 1-element) list. I'll rework the parser to be better, which will also solve your concern that the meaning of `string option` in that constructor is unclear (because it totally is unclear).

Reply:
Returning multiple elements per line seems better indeed!